File: csv_reader.py
import csv
from math import pow

filename = 'input.csv'

entries = {}

# read the CSV file
with open(filename, newline='') as csvfile:
    reader = csv.reader(csvfile)
    next(reader) # skip header

    for row in reader:
        if row[0] not in entries:
            entries[row[0]] = []
        entries[row[0]].append(row[1])

summary = []
for (uuid, values) in entries.items():
    n = 0
    cumsum = 0
    ssq = 0

    # calculate mean
    for value in values:
        n += 1
        cumsum += float(value)
    mean = cumsum / n
    
    # calculate stdev
    for value in values:
        ssq = pow((mean - float(value)), 2)
    stdev = pow((ssq / n), 1./2)

    summary.append([uuid, n, mean, stdev])

# sort by size
summary.sort(key = lambda summary: summary[n], reverse = True)

for [uuid, n, mean, stdev] in summary:
    print("{}\t{}\t{}\t{}".format(uuid, n, mean, stdev))

File: csv_reader.rs
use std::collections::HashMap;
use std::io::BufRead as _;

fn main() {
    let file = std::fs::OpenOptions::new()
        .read(true)
        .open("input.csv")
        .map(|f| std::io::BufReader::new(f))
        .unwrap();

    let mut entries = HashMap::new();
    // read the CSV file
    // do not read the header
    for line in file.lines().skip(1) {
        let line = line.unwrap();
        let mut splits = line.split(",");
        let uuid = splits.next().unwrap().to_owned();
        let value = splits.next().unwrap().parse::<f64>().unwrap();
    
        entries.entry(uuid).or_insert(vec![]).push(value);
    }

    let mut summaries = vec![];
    for (uuid, values) in entries.into_iter() {
        let mut n = 0;
        let mut cumsum = 0.;
        let mut ssq = 0.;

        // calculate mean
        for value in values.iter() {
            n += 1;
            cumsum += *value;
        }
        let mean = cumsum / n as f64;

        // calculate stdev
        for value in values.iter() {
            ssq = (mean - *value).powi(2);
        }
        let stdev = (ssq / n as f64).sqrt();

        summaries.push((uuid, n, mean, stdev));
    }

    // sort by size
    summaries.sort_unstable_by_key(|summary| -summary.1);

    for (uuid, n, mean, stdev) in summaries.into_iter() {
        println!("{}\t{}\t{}\t{}", uuid, n, mean, stdev);
    }
}
